### PR TITLE
Fix error on connection error

### PR DESCRIPTION
**Pull request by dlunch**
Fixes 
```
File "site-packages/kombu/transport/redis_cluster.py", line 372, in parse_response
    self.client.nodes_manager.startup_nodes.pop(target_node.name, None)
NameError: name 'target_node' is not defined
```

Also fixes missing exception catches

### CODE REVIEW

File: kombu/transport/redis_cluster.py [hunk 1]
```diff
@@ -341,7 +341,7 @@ def _brpop_read(self, **options):
             resp = self.parse_response(conn, 'BRPOP', **options)
         except self.connection_errors:
             raise Empty()
-        except MovedError:
+        except (ConnectionError, TimeoutError, MovedError, TryAgainError, ClusterDownError, SlotNotCoveredError, AskError):
```
Comment (bellatoris) [inline kombu/transport/redis_cluster.py:344]: wouldn't it better to add these errors to `self.connection_errors`?

File: kombu/transport/redis_cluster.py [hunk 2]
```diff
@@ -341,18 +341,19 @@ def _brpop_read(self, **options):
...
     def _poll_error(self, cmd, conn, **options):
-        resp = self.parse_response(conn, 'BRPOP', **options)
-        if resp:
-            self.deliver_response(resp)
+        try:
+            resp = self.parse_response(conn, 'BRPOP', **options)
+            if resp:
+                self.deliver_response(resp)
+        except self.connection_errors:
+            pass
```
Comment (bellatoris) [inline kombu/transport/redis_cluster.py:356]: any reason for just passing?

Comment (dlunch) [reply kombu/transport/redis_cluster.py:356]: We should not throw error on `_poll_error` method to continue kombu to work.

Comment (bellatoris) [reply kombu/transport/redis_cluster.py:356]: Ok then let's log exception at least

Comment (bellatoris) [reply kombu/transport/redis_cluster.py:356]: and please add comments why we should not throw an error on that method

**Review by bellatoris — APPROVED** [review]

**Pull request merged**